src/peft/tuners/oft/layer.py

```diff
@@ -189,20 +442,16 @@ def update_layer(self, adapter_name, r, oft_block_size, module_dropout, coft, ep
                 "Something went wrong, please report this error: https://github.com/huggingface/peft/issues"
             )
 
-        self.coft[adapter_name] = coft
-        self.block_share[adapter_name] = block_share
-        self.eps[adapter_name] = eps * math.ceil(self.out_features / r) * math.ceil(self.out_features / r)
-
         # Create weights with provided shape
+        n_elements = oft_block_size * (oft_block_size - 1) // 2
         if block_share:
-            self.oft_r[adapter_name] = nn.Parameter(
-                torch.empty(1, math.ceil(self.in_features / r), math.ceil(self.in_features / r))
+            self.oft_R[adapter_name] = OFTRotationModule(
```

Comment:
Would this change still allow to load old checkpoints and produce the same results or is it backwards incompatible?